File: src/character_count.py
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/character_count', methods=['POST'])
def character_count():
    data = request.get_json()
    text = data.get('text', '')
    total_characters = len(text)
    characters_without_spaces = len(text.replace(' ', ''))
    unique_characters = len(set(text))

    return jsonify({
        'total_characters': total_characters,
        'characters_without_spaces': characters_without_spaces,
        'unique_characters': unique_characters
    })

if __name__ == '__main__':
    app.run(debug=True)

File: src/characterCount.js
const express = require('express');
const router = express.Router();

router.post('/character-count', (req, res) => {
    const text = req.body.text;
    if (!text) {
        return res.status(400).json({ error: 'Text field is required' });
    }

    const totalCharacters = text.length;
    const charactersWithoutSpaces = text.replace(/\s+/g, '').length;
    const uniqueCharacters = [...new Set(text)].length;

    res.json({
        totalCharacters,
        charactersWithoutSpaces,
        uniqueCharacters
    });
});

module.exports = router;